packages/common/src/downloads/_internal/file-url-fetchers/fetchHubApiDownloadFile.ts

```diff
@@ -97,44 +114,77 @@ function getDownloadApiRequestUrl(options: IFetchDownloadFileOptions) {
  * Polls the Hub Download API until the download is ready, then returns the download file URL
  *
  * @param requestUrl Hub Download Api URL to poll
+ * @param pollInterval interval in milliseconds to poll for job completion
+ * @param cacheQueryParam an optional query param to control cache behavior. 'updateCache' should
+ * only be used with the initial request when a download cache update is desired. All subsequent
+ * requests should use'trackCacheUpdate' to follow the progress of the cache update.
+ * for subsequent requests when polling the progress of a cache update.
  * @param progressCallback an optional callback to report download generation progress
  * @returns the final file URL
  */
 async function pollDownloadApi(
   requestUrl: string,
   pollInterval: number,
+  cacheQueryParam?: CacheQueryParam,
   progressCallback?: downloadProgressCallback
 ): Promise<IFetchDownloadFileResponse> {
-  const response = await fetch(requestUrl);
+  // If requested, append the cache query param to the request URL
+  let withCacheQueryParam = requestUrl;
+  if (cacheQueryParam) {
+    withCacheQueryParam = `${requestUrl}&${cacheQueryParam}=true`;
+  }
+
+  const response = await fetch(withCacheQueryParam);
+
   if (!response.ok) {
     const errorBody = await response.json();
     // TODO: Add standarized messageId when available
     throw new ArcgisHubDownloadError({
       rawMessage: errorBody.message,
     });
   }
-  const { status, progressInPercent, resultUrl }: IHubDownloadApiResponse =
-    await response.json();
+
+  const {
+    status,
+    progressInPercent,
+    resultUrl,
+    cacheStatus,
+  }: IHubDownloadApiResponse = await response.json();
+
   const operationStatus = toDownloadOperationStatus(status);
   if (operationStatus === DownloadOperationStatus.FAILED) {
     throw new HubError(
       "fetchHubApiDownloadFileUrl",
       "Download operation failed with a 200"
     );
   }
-  progressCallback && progressCallback(operationStatus, progressInPercent);
 
-  // Operation complete, return the download URL
   if (resultUrl) {
+    // Operation complete, return the download URL and cache status
     return {
       type: "url",
       href: resultUrl,
+      cacheStatus,
     };
   }
 
-  // Operation still in progress, poll again
+  // Operation still in progress. Report progress if a callback was provided and poll again.
+  progressCallback && progressCallback(operationStatus, progressInPercent);
```

Comment:
Moving this for the same reason as above. We don't want to emit the "complete" event from within this function.